routes/phrasesRoute.js

```diff
@@ -0,0 +1,91 @@
+const express = require('express');
+const phrases = require('../phrases');
+const sample = require('lodash.sample');
+const pick = require('lodash.pick');
+
+const routes = express.Router();
+
+const gameState = {};
+let counterID = 0;
+const gameStateKeys = ["id", "category", "failsCounter", "encodedPhrase"];
+
+const createNewGame = () => {
+    let incrementedID = counterID++;
+    const newPhrase = getRandomPhrase();
+    console.log("newphrase", newPhrase);
+    gameState[incrementedID] = {
+        id: incrementedID,
+        category: newPhrase.category,
+        phrase: newPhrase.phrase,
+        failsCounter: 0,
+        encodedPhrase: encryptPhrase(newPhrase.phrase),
+    }
+    return gameState[incrementedID];
+}
+
+const getRandomPhrase = () => {
+    const randomCategory = sample(Object.keys(phrases));
+    const upperPhrase = sample(phrases[randomCategory]).toUpperCase();
+    return {
+        category: randomCategory,
+        phrase: upperPhrase,
+    };
+}
+
+const ommitedChars = [' ', '-'];
+
+const encryptPhrase = (phrase) => {
+    const encryptedString = phrase.split('').map(letter => {
+        if(ommitedChars.includes(letter)) {
+            return letter;
+        }
+        return '_';
+    }).join('');
+    return encryptedString;
+}
+
+routes.get('/new', (req, res) => {
+    const newGame = createNewGame();
+    const reducedNewGame = pick(newGame, gameStateKeys);
+    console.log("newGame", reducedNewGame);
+    res.send(reducedNewGame);
+});
+
+const  revealLetterInPhrase = (fullPhrase, partPhrase, letter) => {
+    const phraseSplitted = fullPhrase.split('');
+    const partPhraseSplitted = partPhrase.split('');
+    console.log("partPhrase", partPhrase)
+    const revealedLetterInPhrase = phraseSplitted
+        .map((element, index) => {
+            const revealedLetters = (element === letter) ? letter : partPhraseSplitted[index];
```

Comment:
Name should be revealedLetter, singular - as you return one element;

Reply:
Right.